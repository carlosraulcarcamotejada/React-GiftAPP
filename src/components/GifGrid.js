import React from 'react';
import useFetchGifs from '../CustomHooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    

    return (
        <>
            <h3> {category}  </h3>
            { loading && <p >Loading...</p>}
            {
                images.map(image =>
                    <GifItem
                        key={image.id}
                        {...image}
                    />)
            }

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}