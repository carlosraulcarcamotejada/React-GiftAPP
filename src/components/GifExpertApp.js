import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'




export const GifExpertApp = ({ categoriesDefault = [] }) => {

    const [categories, setCategories] = useState(categoriesDefault)



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />

            {
                categories.map(category => {
                    return (<GifGrid key={category} category={category} />);
                })
            }

        </>
    )
}

