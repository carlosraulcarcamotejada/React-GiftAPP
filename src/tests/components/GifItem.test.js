import { shallow } from 'enzyme';
import React from 'react';
import { GifItem } from '../../components/GifItem'




describe('Test en el componenten <GifItem />', () => {


    const title = 'un t√≠tulo';
    const url = 'algo';
    const wrapper = shallow(<GifItem url={url} title={title} />);

    test('Prueba que el componente se muestra correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('Prueba 2', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    

});

