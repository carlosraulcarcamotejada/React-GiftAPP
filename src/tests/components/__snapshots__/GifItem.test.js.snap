// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en el componenten <GifItem /> Prueba que el componente se muestra correctamente 1`] = `
<div>
  <img
    alt="un título"
    src="algo"
  />
  <p>
     
    un título
     
  </p>
</div>
`;
